src/app/workflow/launch-third-party/multi-cloud-launch/multi-cloud-launch.component.ts

```diff
@@ -0,0 +1,131 @@
+import { Component, Input, OnInit, ViewChild } from '@angular/core';
+import { MatMenuTrigger } from '@angular/material/menu';
+import { Base } from '../../../shared/base';
+import { CloudInstance, User } from '../../../shared/openapi';
+
+@Component({
+  selector: 'app-multi-cloud-launch',
+  templateUrl: './multi-cloud-launch.component.html',
+  styleUrls: ['./multi-cloud-launch.component.scss'],
+})
+export class MultiCloudLaunchComponent extends Base implements OnInit {
+  @ViewChild(MatMenuTrigger) trigger: MatMenuTrigger;
+
+  @Input()
+  cloudInstances: Array<CloudInstance>;
+
+  @Input()
+  usersCloudInstances: Array<CloudInstance>;
+
+  @Input()
+  languagePartner: CloudInstance.PartnerEnum;
+
+  @Input()
+  launchUrlAppend: string;
+
+  @Input()
+  user: User;
+  _customLaunchWithOption: string;
```

Comment:
nothing particularly wrong with using underscore prefix for private variables but we haven't really be using that convention (same with most code on the Angular site)